src/Features/Core/Portable/ExtractMethod/MethodExtractor.Analyzer.cs

```diff
@@ -753,15 +707,45 @@ private bool SelectionContainsOnlyIdentifierWithSameType(ITypeSymbol type)
             return type.Equals(SelectionResult.GetContainingScopeType());
         }
 
-        protected virtual ITypeSymbol? GetSymbolType(SemanticModel model, ISymbol symbol)
-            => symbol switch
+        private ITypeSymbol? GetSymbolType(SemanticModel semanticModel, ISymbol symbol)
+        {
+            var type = symbol switch
             {
                 ILocalSymbol local => local.Type,
                 IParameterSymbol parameter => parameter.Type,
-                IRangeVariableSymbol rangeVariable => GetRangeVariableType(model, rangeVariable),
+                IRangeVariableSymbol rangeVariable => GetRangeVariableType(semanticModel, rangeVariable),
                 _ => throw ExceptionUtilities.UnexpectedValue(symbol)
             };
 
+            if (type is null)
+                return type;
+
+            // Check if null is possibly assigned to the symbol. If it is, leave nullable annotation as is, otherwise we
+            // can modify the annotation to be NotAnnotated to code that more likely matches the user's intent.
+
+            if (type.NullableAnnotation is not NullableAnnotation.Annotated)
+                return type;
+
+            var selectionOperation = semanticModel.GetOperation(SelectionResult.GetContainingScope());
+
+            // Need to analyze from the start of what we're extracting to the end of the scope that this variable could
+            // have been referenced in.
+            var span = TextSpan.FromBounds(SelectionResult.FinalSpan.Start, SelectionResult.GetContainingScope().Span.End);
+
+            // For Extract-Method we don't care about analyzing the declaration of this variable. For example, even if
+            // it was initially assigned 'null' for the purposes of determining the type of it for a return value, all
+            // we care is if it is null at the end of the selection.  If it is only assigned non-null values, for
+            // example, we want to treat it as non-null.
+            if (selectionOperation is not null &&
+                NullableHelpers.IsSymbolAssignedPossiblyNullValue(
```

Comment:
we now call this existing helper, which also took on some of hte logic i wrote in the original EM helper from a few prs ago.